packages/@jsii/spec/lib/assembly.ts

```diff
@@ -357,6 +357,11 @@ export interface Docs {
    */
   default?: string;
 
+  /**
+   * Hints provided to the compiler via TSDoc annotations.
+   */
+  hints?: TypeSystemHints;
```

Comment:
Does this type hint even need to persist in the assembly?

It does not add information, and the only thing its presence here will do is allow downstream consumers to make mistakes, because they can now accidentally look at `docs.hints.struct` instead of `datatype`, which they should hav ebeen.

Reply:
+1 feels like a leak